src/controllers/v1/task/create.controller.ts

```diff
@@ -0,0 +1,71 @@
+import { z } from 'zod';
+import { DateTime } from 'luxon';
+import { RequestHandler } from 'express';
+
+import { Task, TaskPriority } from '../../../repositories/task/create';
+import { TaskService } from '../../../services/task.service';
+import httpStatus from 'http-status';
+import { logger } from '../../../logger/logger';
+
+export enum NotificationSubscription {
+  reminder = 'reminder',
+  taskStarted = 'taskStarted',
+  taskCompleted = 'taskCompleted'
+}
+
+export const NotifSubValidationSchema = z
+  .object({
+    reminder: z.boolean().optional(),
+    taskStarted: z.boolean().optional(),
+    taskCompleted: z.boolean().optional()
+  })
+  .refine((data) => Object.values(data).some((value) => value === true), {
+    message: `At least one field in [${Object.keys(NotificationSubscription)}] must be set if notificationSubscription is supplied`
+  });
+
+const optionalDateValidationSchema = z
+  .string()
+  .transform((val) => DateTime.fromISO(val))
+  .optional();
+
+export const TaskCreateValidationSchema = z.object({
+  title: z.string().min(3).max(64),
+  description: z.string().max(500).optional(),
+  startTime: optionalDateValidationSchema,
+  completionTime: optionalDateValidationSchema,
+  priorityLevel: z.nativeEnum(TaskPriority).optional(),
+  notificationSubscription: NotifSubValidationSchema.optional()
+});
+
+export type NotificationSubscriptionSchema = z.infer<typeof NotifSubValidationSchema>;
+
+interface ResponseBody<T> {
+  success: boolean;
+  message: string;
+  data: T;
+}
+
+export type TaskCreatePayload = z.infer<typeof TaskCreateValidationSchema>;
+
+export const createTaskHandler: RequestHandler<
+  unknown,
+  ResponseBody<Task>,
+  TaskCreatePayload
+> = async (req, res, next) => {
+  try {
+    const taskService = new TaskService();
```

Comment:
This is not optimal as it is created per request. Consider creating and reusing the instance